src/System.CommandLine/Argument{T}.cs

```diff
@@ -18,17 +18,17 @@ public Argument() : base(null)
             ArgumentType = typeof(T);
         }
 
-        public Argument(string name, Func<T> defaultValue) : this(name)
+        public Argument(string name, Func<T> getDefaultValue) : this(name)
         {
-            SetDefaultValue(() => defaultValue());
+            SetDefaultValueFactory(() => getDefaultValue());
         }
 
         public Argument(Func<T> defaultValue) : this()
         {
-            SetDefaultValue(() => defaultValue());
+            SetDefaultValueFactory(() => defaultValue());
```

Comment:
@jonsequitur: it seems the original bug I hit isn't being fixed here: if I pass null in for defaultValue, this will still call SetDefaultValueFactory and ultimately null ref once the property tries being accessed.

Reply:
Thanks! I realized I made this change on another branch where I'm refactoring toward fixing #723. These APIs may change significantly by the time that's done but the refactorings are worth merging earlier which will actually fix this issue.